### PR TITLE
Improve lis3dh driver.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/lis3dh.toit [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    // We are always in high-resolution mode:
+    // - RANGE_2G:   1mg/LSB
+    // - RANGE_4G:   2mg/LSB
+    // - RANGE_8G:   4mg/LSB
+    // - RANGE_16G:  12mg/LSB   // <- Note that the 16G sensitivity is not 8mg/LSB as expected.
+
+    if range_ != RANGE_16G:
+      // It just happens that RANGE_2G to RANGE_8G can be calculated this way.
+      // We could also make 4 ifs.
+      x *= range_ + 1
```
Comment (erikcorry) [inline src/lis3dh.toit]: Should be shift?

**Review by erikcorry — APPROVED** [review]

File: src/lis3dh.toit [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    z := reg_.read_i16_le (OUT_Z_L_ | AUTO_INCREMENT_BIT)
+
+    // Section 2.1, table4:
+    // The linear acceleration sensitivity depends on the range.
+    // We are always in high-resolution mode:
+    // - RANGE_2G:   1mg/LSB
+    // - RANGE_4G:   2mg/LSB
+    // - RANGE_8G:   4mg/LSB
+    // - RANGE_16G:  12mg/LSB   // <- Note that the 16G sensitivity is not 8mg/LSB as expected.
+
```
Comment (erikcorry) [inline src/lis3dh.toit]:
```suggestion
multiplier := #[2, 4, 8, 12][range_]
x *= multiplier
y *= multiplier
z *= multiplier
```

**Review by erikcorry — APPROVED** [review]

File: src/lis3dh.toit (hunk 1 above)
Comment (floitsch) [reply src/lis3dh.toit]: done.

File: src/lis3dh.toit (hunk 2 above)
Comment (floitsch) [reply src/lis3dh.toit]: done.

**Pull request merged**